tests/test_parser.py

```diff
@@ -0,0 +1,76 @@
+import os
+import sys
+
+THIS_FOLDER = os.path.dirname(os.path.abspath(__file__)) + "/.."
+sys.path.insert(1, THIS_FOLDER)
+
+from vyxal.lexer import *
+from vyxal.parser import *
+
+
+def fully_parse(program: str) -> list[Structure]:
+    """
+    Essentially, wrap tokenise(program) in parse
+
+    Parameters
+    ----------
+
+    program : str
+        The program to tokenise and then parse
+
+    Returns
+    -------
+
+    list[Structure]
+        Quite literally parse(tokenise(program))
+    """
+
+    return parse(tokenise(program))  # see what I mean?
+
+
+def test_basic():
+    assert (
+        str(fully_parse("1 1+"))
+        == "[['none', ['number', '1']], ['none', ['number', '1']], "
+        + "['none', ['general', '+']]]"
+    )
+
+    assert (
+        str(fully_parse("`Hello, World!`"))
+        == "[['none', ['string', " + "'Hello, World!']]]"
+    )
+
+
+def test_fizzbuzz():
+    assert eval(str(fully_parse("₁ƛ₍₃₅kF½*∑∴"))) == [
+        ["none", ["general", "₁"]],
+        [
+            "lambda",
+            [
+                "1",
+                [
+                    [
+                        "dyadic_modifier",
+                        [
+                            "₍",
+                            [
+                                ["none", ["general", "₃"]],
+                                ["none", ["general", "₅"]],
+                            ],
+                        ],
+                    ],
+                    ["none", ["general", "kF"]],
+                    ["none", ["general", "½"]],
+                    ["none", ["general", "*"]],
+                    ["none", ["general", "∑"]],
+                    ["none", ["general", "∴"]],
+                ],
+            ],
+        ],
+        ["none", ["general", "M"]],
+    ]
+
+
+if __name__ == "__main__":  # For testing outside of the workflow
```

Comment:
what exactly is "the workflow"? Using `pytest`? Just use pytest; it's not very complicated

Reply:
Yeah, the tests should simply be functions whose names start with `test_`. We shouldn't need the `__name__ == "__main__"` thing *anywhere* except in the main interpreter file.

Reply:
Outside of "the workflow" is testing locally using the command line.

Reply:
@Lyxal It's still fine - we can do that using `pytest tests` (or whatever directory/file you want).

Reply:
I more mean actually running the file on my computer, because I don't have py_tests set up locally.